src/main/java/bussiness_layer/utils/CourseDtoValidator.java

```diff
@@ -0,0 +1,85 @@
+package bussiness_layer.utils;
+
+import bussiness_layer.dto.CourseDto;
+import bussiness_layer.dto.LessonTemplateDto;
+import lombok.experimental.UtilityClass;
+import utils.LessonType;
+import utils.exceptions.UnprocessableEntityException;
+
+import java.util.List;
+
+@UtilityClass
+public class CourseDtoValidator {
+    public static void validate(CourseDto courseDto){
```

Comment:
No, I don't think it's a good idea, because validate(List<Entity>) means we validate each Entity independently. Here we validate that List as a whole. I'm expecting actually (though I know this will never happen) that other developers will add here other validations for CourseDto.
We actually have another problem: we have validations made in 2 places - here and those made by spring/lombok. I think it will be better if we either find a solution to make the framework validate this things for us, or remove the validator annotations from dtos and have all validations in an utility class like this

Reply:
It's ok to have a validator for a class/dto and to also use spring validations on that class/dto, because there are some cases (like this) when spring validations are not enough. I think it's almost impossible to make all the validations you have done in validator class using just annotations. A solution would be to break down the lessonTemplateDtoList in course dto in small lists, one for each type, but by doing this we will have to refactor a lot in the code. Also, I'm not sure this modification will ensure us that all the validations will be done by just using spring ( for example, I don't think we can check for the number of seminars and partial seminars using a SINGLE annotation). If you intend to validate other fields in a course dto passed as a parameter to this function, then it's okay to pass all the dto, I think that the other fields in course dto could be validated just by spring. What do you think?